Magento_Theme/web/js/select.babel.js

```diff
@@ -6,17 +6,46 @@ define([
   'use strict'
 
   return function (config, select) {
+    const isSearchable = $(select).hasClass('select--search')
+
+    function matchCustom(params, data) {
+      if ($(data.element).data('option') === 'fallback') {
+        return data;
+      }
+
+      if ($.trim(params.term) === '') {
+        return data;
+      }
+
+      if (typeof data.text === 'undefined') {
+        return null;
+      }
+
+      if (data.text.toLowerCase().indexOf(params.term.toLowerCase()) > -1) {
+        return data;
+      }
+
+      return null;
+    }
+
     const selectId = select.id,
       defaultOptions = {
-        minimumResultsForSearch: Infinity,
+        minimumResultsForSearch: isSearchable ? undefined : Infinity,
         width: null,
         position: 'bottom',
         debug: true
       }
```

Comment:
what do you think about moving all searchable select options (`dropdownCssClass`, `matcher`) here, similarly to `minimumResultsForSearch`?